x-pack/legacy/plugins/monitoring/public/np_imports/angular/index.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import angular, { IModule } from 'angular';
+
+import {
+  AppMountContext,
+  LegacyCoreStart
+} from 'kibana/public';
+
+import { configureAppAngularModule } from '../legacy_imports';
+
+//@ts-ignore TODO: change to absolute path
+import uiRoutes from 'plugins/monitoring/np_imports/ui/routes';
+//@ts-ignore TODO: change to absolute path
+import { uiModules } from 'plugins/monitoring/np_imports/ui/modules';
+//@ts-ignore TODO: change to absolute path
+import { registerTimefilterWithGlobalState } from 'plugins/monitoring/np_imports/ui/timefilter';
+
+import { localAppModule, appModuleName } from './modules';
+
+export class AngularApp {
+
+  private _injector?: angular.auto.IInjectorService;
+  private _observer?: MutationObserver;
+
+  constructor({ core }: AppMountContext, { appBasePath, element }: { appBasePath: string, element: HTMLElement }) {
+
+    const inDOM = () => document.body.contains(element);
+    if (inDOM()) {
+      this.setup(core, element, appBasePath);
+      return;
+    }
+
+    this._observer = new MutationObserver(() => {
+      if (inDOM()) {
+        this._observer!.disconnect();
+        this._observer = undefined;
+        this.setup(core, element, appBasePath);
+      }
+    });
+
+    this._observer.observe(document.body, { childList: true, subtree: true });
```

Comment:
Why we need to observe all updates down the tree?

Reply:
This was part of: https://github.com/elastic/kibana/pull/51823#discussion_r352678135 I just checked and it seems like the element is already in DOM initially. I'll go ahead and remove this check/logic